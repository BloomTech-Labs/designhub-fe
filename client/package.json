{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@auth0/auth0-spa-js": "^1.2.3",
    "@sentry/browser": "^5.6.3",
    "apexcharts": "^3.8.6",
    "axios": "^0.19.0",
    "dotenv": "^8.1.0",
    "node-sass": "^4.12.0",
    "react": "^16.9.0",
    "react-apexcharts": "^1.3.3",
    "react-calendar-heatmap": "^1.8.1",
    "react-dom": "^16.9.0",
    "react-redux": "^7.1.1",
    "react-router-dom": "^5.0.1",
    "react-scripts": "3.1.1",
    "react-tabs": "^3.0.0",
    "redux": "^4.0.4",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.3.0",
    "uuid": "^3.3.3"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --verbose",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^4.1.0",
    "@testing-library/react": "^9.1.4",
    "prettier": "^1.18.2",
    "react-test-renderer": "^16.9.0"
  }
}
