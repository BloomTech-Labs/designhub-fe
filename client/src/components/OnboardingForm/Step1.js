import React from 'react';

const Step1 = ({ formUser, onChange }) => {
  const {
    bio,
    email,
    firstName,
    lastName,
    location,
    username,
    website
  } = formUser;
  return (
    <>
      <label htmlFor="firstName">First Name</label>
      <input
        id="firstName"
        name="firstName"
        type="text"
        value={firstName}
        onChange={onChange}
      />
      <label htmlFor="lastName">Last Name</label>
      <input
        id="lastName"
        name="lastName"
        type="text"
        value={lastName}
        onChange={onChange}
      />
      <label htmlFor="username">Username</label>
      <input
        required
        id="username"
        name="username"
        type="text"
        value={username}
        onChange={onChange}
      />
      <label htmlFor="email">Email</label>
      <input
        required
        id="email"
        name="email"
        type="text"
        value={email}
        onChange={onChange}
      />
      <label htmlFor="bio">Description</label>
      <textarea
        cols="30"
        rows="4"
        name="bio"
        id="bio"
        value={bio}
        onChange={onChange}
      />
      <label htmlFor="location">Location</label>
      <input
        id="location"
        name="location"
        type="text"
        value={location}
        onChange={onChange}
      />

      <label htmlFor="website">Website</label>
      <input
        id="website"
        name="website"
        type="text"
        value={website}
        onChange={onChange}
      />
    </>
  );
};

export default Step1;
