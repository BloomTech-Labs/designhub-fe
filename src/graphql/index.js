/*
  QUERIES
*/

export { default as GET_PROJECT_BY_ID_QUERY } from './queries/projects/project';
export { default as GET_ALL_PROJECTS_QUERY } from './queries/projects/projects';
export { default as GET_USER_BY_ID_QUERY } from './queries/users/user';
export { default as GET_ALL_USERS_QUERY } from './queries/users/users';

/*
MUTATIONS
*/

export { default as ADD_COMMENT_MUTATION } from './mutations/comments/addComments';
// export { default as ADD_PHOTO_COMMENT_MUTATION } from './mutations/comments/addPhotoComments';
export { default as ADD_PROJECT_MUTATION } from './mutations/projects/addProject';
export { default as ADD_PROJECT_PHOTO_MUTATION } from './mutations/projects/addProjectPhoto';
export { default as ADD_USER_MUTATION } from './mutations/users/addUser';
// export { default as ADD_CATEGORY_MUTATION } from './mutations/categories/addCategory';
// export { default as ADD_HEATMAP_MUTATION } from './mutations/heatmap/addHeatmap';
// export { default as ADD_INVITE_MUTATION } from './mutations/invites/addInvite';
// export { default as ADD_INVITE_COMMENTS_QUERY } from './mutations/invites/addInviteComments';
// export { default as ADD_INVITE_FOLLOW_MUTATION } from './mutations/invites/addInviteFollow';
// export { default as ADD_INVITE_STARED_MUTATION } from './mutations/invites/addInviteStarred';
export { default as ADD_FOLLOWER_MUTATION } from './mutations/users/addFollower';
// export { default as ADD_STARED_MUTATION } from './mutations/users/addStarred';
// export { default as ADD_USER_RESEARCH_MUTATION } from './mutations/users/addUserResearch';
// export { default as ADD_USER_RESEARCHING_MUTATION } from './mutations/users/addUserResearching';
// export { default as UPDATE_CATEGORY_MUTATION } from './mutations/categories/updateCategory';
// export { default as UPDATE_COMMENT_MUTATION } from './mutations/comments/updateComments';
// export { default as UPDATE_PHOTO_COMMENT_MUTATION } from './mutations/comments/updatePhotoComments';
export { default as UPDATE_PROJECT_MUTATION } from './mutations/projects/updateProject';
export { default as UPDATE_USER_MUTATION } from './mutations/users/updateUser';
// export { default as UPDATE_INVITE_MUTATION } from './mutations/invites/updateInvites';
// export { default as UPDATE_PROJECT_INVITE_MUTATION } from './mutations/invites/updateProjectInvite';
// export { default as UPDATE_PROJECT_INVITES_MUTATION } from './mutations/invites/updateProjectInvites';
export { default as UPDATE_PROJECT_PHOTO_MUTATION } from './mutations/projects/updateProjectPhoto';
// export { default as DELETE_HEATMAP_MUTATION } from './mutations/heatmap/deleteHeatmap';
// export { default as DELETE_PHOTO_COMMENT_MUTATION } from './mutations/comments/deletePhotoComments';
export { default as DELETE_PROJECT_PHOTO_MUTATION } from './mutations/projects/deleteProjectPhoto';
// export { default as DELETE_USER_RESEARCH_MUTATION } from './mutations/users/deleteUserResearch';
// export { default as DELETE_STARED_MUTATION } from './mutations/users/deleteStarred';
// export { default as DELETE_PROJECT_INVITE_MUTATION } from './mutations/invites/deleteProjectInvite';
// export { default as DELETE_CATEGORY_MUTATION } from './mutations/categories/deleteCategory';
export { default as DELETE_FOLLOWER_MUTATION } from './mutations/users/deleteFollower';
export { default as DELETE_PROJECT_MUTATION } from './mutations/projects/deleteProject';
// export { default as DELETE_INVITE_MUTATION } from './mutations/invites/deleteInvite';
export { default as DELETE_COMMENTS_MUTATION } from './mutations/comments/deleteComments';
export { default as DELETE_USER_MUTATION } from './mutations/users/deleteUser';
export { default as SEARCH_MUTATION } from './mutations/users/search';
