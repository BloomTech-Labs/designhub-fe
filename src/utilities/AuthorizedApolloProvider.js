import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import { setContext } from '@apollo/link-context';
import React from 'react';

import { useAuth0 } from './auth-spa';

const AuthorizedApolloProvider = ({ children }) => {
  const { getTokenSilently } = useAuth0();

  const httpLink = createHttpLink({
    uri: process.env.REACT_APP_DOMAIN,
  });

  const authLink = setContext(async () => {
    const token = await getTokenSilently();
    return {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    };
  });

  const apolloClient = new ApolloClient({
    link: authLink.concat(httpLink),
    cache: new InMemoryCache(),
    connectToDevTools: true,
  });

  return <ApolloProvider client={apolloClient}>{children}</ApolloProvider>;
};

export default AuthorizedApolloProvider;
