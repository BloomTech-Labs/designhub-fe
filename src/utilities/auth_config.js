const config = {
  domain: `${process.env.REACT_APP_DOMAIN}`,
  clientId: `${process.env.REACT_APP_CLIENT_ID}`,
  audience: `${process.env.REACT_APP_BASE_URL}`,
};

export default config;
