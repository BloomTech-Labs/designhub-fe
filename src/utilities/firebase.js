import * as firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import "firebase/storage";
// firebase app init and config
const firebaseConfig = {
  apiKey: process.env.REACT_APP_API_KEY,
  authDomain: process.env.REACT_APP_AUTH_DOMAIN,
  databaseURL: process.env.REACT_APP_DATABASE_URL,
  projectId: process.env.REACT_APP_PROJECT_ID,
  storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
  appId: process.env.REACT_APP_APP_ID,
};
firebase.initializeApp(firebaseConfig);
firebase.firestore();
export const storage = firebase.storage();
export default firebase;