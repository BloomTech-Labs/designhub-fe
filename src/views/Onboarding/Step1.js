import React, { useState } from 'react';

import welcome from '../../ASSETS/welcome.svg';

import CharacterCount from '../../common/CharacterCount/CharacterCount';

const Step1 = ({
  alert,
  formUser,
  onChange,
  showPrev,
  handleNextButton,
  handlePrevButton,
  logout,
  submitButton,
}) => {
  const {
    bio,
    //email,
    firstName,
    lastName,
    location,
    username,
    website
  } = formUser;

  const [firstNameRef, setFirstNameRef] = useState(null);
  const [lastNameRef, setLastNameRef] = useState(null);
  const [usernameRef, setUsernameRef] = useState(null);
 // const [emailRef, setEmailRef] = useState(null);

  const focusRef = thisRef => {
    thisRef.scrollIntoView({ behavior: 'smooth', block: 'center' });
  };

  let userNameAlertClass = 'required input-field';
  if (alert.username) {
    alert.username === 'taken'
      ? (userNameAlertClass = 'required alert taken input-field')
      : (userNameAlertClass = 'required alert input-field');
    focusRef(usernameRef);
  }
 // if (alert.email) focusRef(emailRef);
  if (alert.lastName) focusRef(lastNameRef);
  if (alert.firstName) focusRef(firstNameRef);

  return (
    <div className="FormStep">
      <div className="left-side">
        <h6 className="steps">Step 1 / 2</h6>
        <div className="left-container">
          <header>
            <h1>Let's get to know you more!</h1>
            <p>This information will be visible on your profile.</p>
          </header>
          <div className="first_lastname-flex">
            <div>
              <div
                className={
                  alert.firstName
                    ? 'required alert input-field'
                    : 'required input-field'
                }
              >
                <label htmlFor="firstName">First Name</label>
                <input
                  required
                  id="firstName"
                  name="firstName"
                  type="text"
                  value={firstName}
                  onChange={onChange}
                  placeholder="i.e. Erik"
                  ref={setFirstNameRef}
                />
              </div>
            </div>

            <div>
              <div
                className={
                  alert.lastName
                    ? 'required alert input-field'
                    : 'required input-field'
                }
              >
                <label htmlFor="lastName">Last Name</label>
                <input
                  required
                  id="lastName"
                  name="lastName"
                  type="text"
                  value={lastName}
                  onChange={onChange}
                  placeholder="i.e. Lambert"
                  ref={setLastNameRef}
                />
              </div>
            </div>
          </div>

          <div>
            <div className={userNameAlertClass}>
              <label htmlFor="username">Username</label>
              <input
                required
                id="username"
                name="username"
                type="text"
                value={username}
                onChange={onChange}
                placeholder="i.e. eriklambert"
                ref={setUsernameRef}
              />
            </div>
          </div>

          <div>
{/*            <div
              className={
                alert.email
                  ? 'required alert input-field'
                  : 'required input-field'
              }
            >
              <label className="req" htmlFor="email">
                Email
              </label>
              <input
                required
                id="email"
                name="email"
                type="text"
                value={email}
                //disabled
                onChange={onChange}
                placeholder="i.e. eriklambert@designhubx.com"
                ref={setEmailRef}
              />
            </div>*/}
          </div>
          <div className="input-field">
            <label className="bio" htmlFor="bio">
              Bio
            </label>
            <textarea
              cols="30"
              rows="4"
              name="bio"
              id="bio"
              value={bio}
              onChange={onChange}
              maxLength="240"
              placeholder="Describe yourself! This will appear on your profile in your bio!"
            />
            <CharacterCount string={bio} limit={240} />
          </div>
          <div className="location-website-flex">
            <div className="input-field">
              <label htmlFor="location">Location</label>
              <input
                id="location"
                name="location"
                type="text"
                value={location}
                onChange={onChange}
                placeholder="i.e. Austin, TX"
              />
            </div>
            <div className="input-field">
              <label htmlFor="website">Website</label>
              <input
                id="website"
                name="website"
                type="text"
                value={website}
                onChange={onChange}
                placeholder="i.e. https://eriklambert.ux"
              />
            </div>
          </div>
          <div className="buttons">
            {showPrev ? (
              <button
                name="prev"
                className="prev-btn"
                onClick={handlePrevButton}
              >
                Previous
              </button>
            ) : (
                <button name="cancel" className="prev-btn" onClick={logout}>
                  Cancel
              </button>
              )}
            {submitButton ? (
              <button className="next-btn" type="submit">
                Submit
              </button>
            ) : (
                <button
                  name="next"
                  className="next-btn"
                  onClick={handleNextButton}
                >
                  Next
              </button>
              )}
          </div>
        </div>
      </div>

      <div className="right-side">
        <header>
          <img src={welcome} alt="welcome to designhub" className="welcome" />
          <h2>
            Create projects and receive feedback by displaying your work on
            DesignHub
          </h2>
        </header>
      </div>
    </div>
  );
};

export default Step1;

