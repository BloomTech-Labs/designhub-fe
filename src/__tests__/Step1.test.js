import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Step1 from '../views/Onboarding/Step1.js';
import { MockedProvider } from '@apollo/client/testing';
import { ApolloProvider } from '@apollo/react-hooks';
import ApolloClient from 'apollo-boost';
import { useAuth0 } from '../utilities/auth-spa';
import { mount, shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { GET_USER_BY_ID_QUERY } from '../graphql';

configure({ adapter: new Adapter() });
const history = createMemoryHistory();
const user = {
  email: 'johndoe@me.com',
  email_verified: true,
  sub: 'google-oauth2|2147627834623744883746',
};

const mocks = [
  {
    request: {
      query: GET_USER_BY_ID_QUERY,
      variables: {
        id: 'google-oauth2|2147627834623744883746',
      },
    },
    result: {
      data: {
        user: {
          id: 'google-oauth2|2147627834623744883746',
          firstName: 'Buck',
          lastName: 'bulldog',
          username: 'buckbulldog',
          email: 'buck@bulldog.com',
          location: 'Buckville, USA',
          bio: 'Do not mess with buck',
          website: 'buckwild.org',
          avatar: 'https://bulldog.img',
        },
      },
    },
  },
];

const client = new ApolloClient({
  uri: process.env.REACT_APP_API_URL,
});

jest.mock('../utilities/auth-spa');
describe('components/Step1 - logged in', () => {
  beforeEach(() => {
    // Mock the Auth0 hook and make it return a logged in state
    useAuth0.mockReturnValue({
      isAuthenticated: true,
      user,
      logout: jest.fn(),
      loginWithRedirect: jest.fn(),
    });
  });
  it('Renders with required props', async () => {
    const wrapper = shallow(
      <ApolloProvider client={client}>
        <Router history={history}>
          <Step1 />
        </Router>
      </ApolloProvider>
    );
    expect(wrapper).toBeTruthy();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
