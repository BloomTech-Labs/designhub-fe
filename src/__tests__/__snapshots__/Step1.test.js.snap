// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Step1 - logged in Renders with required props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
